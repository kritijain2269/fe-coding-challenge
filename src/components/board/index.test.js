import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import { Board } from '.'
import configureStore from '../../store'
import { Provider } from 'react-redux'

test('renders Board', () => {
  const { container } = render(
    <Provider store={configureStore()}>
      <Board />
    </Provider>
  )
  expect(container).toMatchSnapshot()
})

test('should display X on click of cell', () => {
  const { getAllByTestId } = render(
    <Provider store={configureStore()}>
      <Board />
    </Provider>
  )
  const cell = getAllByTestId('cell')
  fireEvent.click(cell[0])
  expect(cell[0].textContent).toEqual('X')
})

test('should display X in atleast one cell on clicking generate random button', () => {
  const { getByTestId, getAllByTestId } = render(
    <Provider store={configureStore()}>
      <Board />
    </Provider>
  )
  const generateRandom = getByTestId('btnGenerateRandom')
  const cell = getAllByTestId('cell')
  fireEvent.click(generateRandom)
  const result = cell.some((x) => x.textContent === 'X')
  expect(result).toEqual(true)
})
