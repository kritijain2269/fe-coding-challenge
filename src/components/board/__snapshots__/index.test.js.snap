// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Board 1`] = `
<div>
  <h1
    class="Heading1"
  >
    Tic Tac Toe
  </h1>
  <div
    class="ButtonWrapper"
  >
    <button
      class="StartButton"
      data-testid="btnStartGame"
    >
      Start new game
    </button>
  </div>
  <div
    class="Board"
  >
    <div
      class="BoardRow"
    >
      <div
        class="Cell CellX"
        data-testid="cell"
      />
      <div
        class="Cell CellX"
        data-testid="cell"
      />
      <div
        class="Cell CellX"
        data-testid="cell"
      />
    </div>
    <div
      class="BoardRow"
    >
      <div
        class="Cell CellX"
        data-testid="cell"
      />
      <div
        class="Cell CellX"
        data-testid="cell"
      />
      <div
        class="Cell CellX"
        data-testid="cell"
      />
    </div>
    <div
      class="BoardRow"
    >
      <div
        class="Cell CellX"
        data-testid="cell"
      />
      <div
        class="Cell CellX"
        data-testid="cell"
      />
      <div
        class="Cell CellX"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="Player"
  >
    <div
      class="NextMove"
    >
      Next move :
      <span
        class="CurrentPlayer CellX"
        data-testid="nextPlayer"
      >
        Player 
        X
      </span>
    </div>
    <div
      class="ButtonWrapper"
    >
      <button
        class="StartButton"
        data-testid="btnGenerateRandom"
      >
        Generate random
      </button>
    </div>
    <span
      class="NoteMessage"
    >
      Note: You can play this game either by clicking on each cell or geneate random button
    </span>
  </div>
</div>
`;
